qiskit/_result.py

```diff
@@ -33,6 +34,8 @@ class Result(object):
     def __init__(self, qobj_result, qobj):
         self.__qobj = qobj
         self.__result = qobj_result
+        if not 'job_id' in self.__result:
```

Comment:
As mentioned on the general comment, this is the `if` that should be dropped, as `job_id` would be present in all `result` dicts.

Reply:
Yes, I agree that it would be better to be more homogeneous and not have this conditional attribute. I was initially unsure about modifying the local backends so I went for minimal changes. 
But I'm happy to do these modifications.